lib/Semantics/mod-file.h

```diff
@@ -33,17 +33,24 @@ class ModFileWriter {
 
 private:
   SemanticsContext &context_;
-  std::stringstream uses_;
-  std::stringstream useExtraAttrs_;  // attrs added to used entity
-  std::stringstream decls_;
-  std::stringstream contains_;
+  // Buffer to use with raw_string_ostream
+  std::string uses_Buf;
+  std::string useExtraAttrs_Buf;
+  std::string decls_Buf;
+  std::string contains_Buf;
```

Comment:
Do these buffers need to be part of this class? Or can they be provided by the calling context and destructed there?

Reply:
As far as I understand yes, it needs to be here. But  I double-check with @DavidTruby and he agrees.

Reply:
If we don't have these here, the strings won't persist across member function calls, and these functions rely on that behaviour sometimes. Otherwise we could just have the streams themselves also not be part of the class.

Reply:
To be clear, I wasn't suggesting the backing strings be constructed and destroyed in some interleaved way. Clearly they need to persist throughout the context for correct output. This seems like a fine place to put them as they'll have the same lifetime as the streams. On the other hand, the object that owns the streams had explicit ways to reset the buffers at one point, which suggested that a single writer object might be reused to generate multiple outputs.